sdk/python/kfp/v2/compiler_cli_tests/test_data/pipeline_with_various_io_types.json

```diff
@@ -144,117 +245,10 @@
         }
       }
     },
-    "schemaVersion": "2.0.0",
-    "sdkVersion": "kfp-1.5.0-rc.3",
-    "root": {
-      "inputDefinitions": {
-        "parameters": {
-          "input1": {
-            "type": "STRING"
-          }
-        },
-        "artifacts": {
```

Comment:
Good to see these invalid definitions gone. Would the compiler still produce them if the pipeline inputs have custom types?

Reply:
That's a good point. I think it's still allowed. Let me double check and throw an error if that's the case.

Reply:
Actually I remember why I had to change this. It won't compile, but the error isn't user friendly. Having artifact at root level will fail compilation not because of an explicit check on it but because of an assertion failure when it ensures a `component_input_parameter` can be found in the root level input parameters list.
Anyway, I'll make a explicit check on this and give a clearer error.